typescript/t.d.ts

```diff
@@ -287,10 +288,16 @@ export interface TFunction<Ns extends Namespace = DefaultNamespace, KPrefix = un
     const ActualOptions extends TOpt & InterpolationMap<Ret> = TOpt & InterpolationMap<Ret>,
     DefaultValue extends string = never,
   >(
-    ...args:
-      | [key: Key | Key[], options?: ActualOptions]
-      | [key: string | string[], options: TOpt & $Dictionary & { defaultValue: DefaultValue }]
-      | [key: string | string[], defaultValue: DefaultValue, options?: TOpt & $Dictionary]
+    ...args: _StrictKeyChecks extends true
+      ? [
+          key: Key | Key[],
+          optionsOrDefaultValue?: ActualOptions | string,
+          options?: TOpt & $Dictionary,
+        ]
+      :
```

Comment:
Maybe there are better ways, but this totally works. So we have to kind of "overload" the second argument. I have tried to do this like below (which is ideal i think), but the types are messed up.

```ts
...args: _StrictKeyChecks extends true
      ? | [key: Key | Key[], options?: ActualOptions]
        | [key: Key | Key[], defaultValue: string; options?: TOpt & $Dictionary]
```